packages/github-auth/node-tests/searcher-test.js

```diff
@@ -5,52 +5,399 @@ const {
 
 describe('github-auth/searcher', function() {
   let env, searchers;
 
-  async function setup() {
-    let factory = new JSONAPIFactory();
-
-    factory.addResource('data-sources', 'github').withAttributes({
-      sourceType: '@cardstack/github-auth',
-      params: {
-        'client-id': 'mock-github-client-id',
-        'client-secret': 'mock-github-client-secret',
-        token: 'mock-github-token'
-      }
-    });
-
-    env = await createDefaultEnvironment(`${__dirname}/github-authenticator`, factory.getModels());
-    searchers = env.lookup('hub:searchers');
+  async function alterExpiration(branch, type, id, interval) {
+    let client = env.lookup(`plugin-client:${require.resolve('@cardstack/pgsearch/client')}`);
+    let result = await client.query('update documents set expires = expires + $1 where branch=$2 and type=$3 and id=$4', [interval, branch, type, id]);
+    if (result.rowCount !== 1) {
+      throw new Error(`test was unable to alter expiration`);
+    }
   }
 
   async function teardown() {
     await destroyDefaultEnvironment(env);
   }
 
-  before(setup);
-  after(teardown);
+  describe('github no permissions configured', function() {
+    async function setup() {
+      let factory = new JSONAPIFactory();
+
+      factory.addResource('data-sources', 'github').withAttributes({
+        sourceType: '@cardstack/github-auth',
+        params: {
+          'client-id': 'mock-github-client-id',
+          'client-secret': 'mock-github-client-secret',
+          token: 'mock-github-token',
+        }
+      });
+
+      env = await createDefaultEnvironment(`${__dirname}/github-authenticator`, factory.getModels());
+      searchers = env.lookup('hub:searchers');
+    }
+
+    beforeEach(setup);
+    afterEach(teardown);
+
+    it('does not set permissions for user when no permissions configured', async function() {
+      let userMock = Object.assign({}, githubUsersResponse);
+      let { login:id } = userMock;
+
+      nock('https://api.github.com')
+        .get(`/users/${id}`)
+        .reply(200, userMock);
+
+      let user = await searchers.get(env.session, 'master', 'github-users', id);
+      expect(user.data.attributes.permissions.length).to.equal(0);
+    });
+  });
+
+  describe('github user permissions', function() {
+    async function setup() {
+      let factory = new JSONAPIFactory();
 
-  it('can get github user', async function() {
-    const { login:id } = githubUsersResponse;
+      factory.addResource('data-sources', 'github').withAttributes({
+        sourceType: '@cardstack/github-auth',
+        params: {
+          'client-id': 'mock-github-client-id',
+          'client-secret': 'mock-github-client-secret',
+          token: 'mock-github-token',
+          permissions: [
+            { repo: 'cardstack/repo1', permission: 'read' },
+            { repo: 'cardstack/repo1', permission: 'write' },
+            { repo: 'cardstack/repo1', permission: 'admin' },
```

Comment:
@ef4 I'm wondering if I made the permissions to complex of an object. maybe instead of representing permissions as an array of objects, it would be easier to setup groups if they were an array of strings? like: 
```js
[ 'cardstack/repo1:read', 'cardstack/repo1:write', 'cardstack/repo1:admin']
```

what do you think?

Reply:
Yeah, seems good. Try to write some actual groups to be sure.